cmd/dlt/imagemirror/cmd.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagemirror
+
+import (
+	"context"
+	"fmt"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	use     = "image-mirror"
+	short   = "Delete image mirror from a cluster"
+	long    = "Delete an image mirror configuration from a ROSA HCP cluster by ID."
+	example = `  # Delete image mirror with ID "abc123" from cluster "mycluster"
+  rosa delete image-mirror --cluster=mycluster abc123
+
+  # Delete without confirmation prompt
+  rosa delete image-mirror --cluster=mycluster abc123 --yes
+
+  # Alternative: using the --id flag
+  rosa delete image-mirror --cluster=mycluster --id=abc123`
+)
+
+var (
+	aliases = []string{"image-mirrors"}
+	args    DeleteImageMirrorArgs
+)
+
+type DeleteImageMirrorArgs struct {
+	Id  string
+	Yes bool
+}
+
+func NewDeleteImageMirrorCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   short,
+		Long:    long,
+		Aliases: aliases,
+		Example: example,
+		Args:    cobra.MaximumNArgs(1),
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), DeleteImageMirrorRunner()),
+	}
+
+	flags := cmd.Flags()
+
+	flags.StringVar(
+		&args.Id,
+		"id",
+		"",
+		"ID of the image mirror configuration to delete",
+	)
+
+	flags.BoolVarP(
+		&args.Yes,
+		"yes",
+		"y",
+		false,
+		"Automatically answer yes to confirm deletion",
+	)
+
+	ocm.AddClusterFlag(cmd)
+	return cmd
+}
+
+func DeleteImageMirrorRunner() rosa.CommandRunner {
```

Comment:
Could we move the code within the runners into a `pkg/` file? 

That way the command logic is in one location, and used in the commands- the `machinepool` commands follow this structure (along with the `userOptions` / `options` design for arg validation)